collectors/modality-probe-debug-collector/src/lib.rs

```diff
@@ -399,20 +402,40 @@ impl Collector {
     /// Collect all new logs, return a report
     pub fn collect_report(&mut self) -> Result<Option<Report>, Error> {
         self.rbuf.clear();
+
         let num_missed = self.reader.read(&mut self.rbuf)?;
         // Possibly add entries missed event
         if num_missed > 0 {
             let num_missed_rounded = u64::min(num_missed, u32::MAX as u64) as u32;
             let (ev, payload) =
                 LogEntry::event_with_payload(EventId::EVENT_LOG_ITEMS_MISSED, num_missed_rounded);
             self.rbuf.insert(0, WholeEntry::Double(ev, payload));
+
+            if let Some(e) = self.prev_paired_wall_clock_time.take() {
+                eprintln!("Warning: dropping paired wall clock time entry {:?} from previous report because items were missed and the associated entry was lost", e);
+            }
+        } else {
+            // Insert paired wall clock time entry from previous report if needed
+            if let Some(e) = self.prev_paired_wall_clock_time.take() {
```

Comment:
Can collapse this into the above `else` clause. I'm a little surprised `clippy` isn't demanding as much.